templates/api/deployment.yaml

```
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "farmdoc.fullname" . }}-api
+  labels:
+    {{- include "farmdoc.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.api.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "farmdoc.name" . }}-api
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "farmdoc.name" . }}-api
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      {{- with .Values.api.image.pullSecrets | default .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: api
+          image: "{{ .Values.image.project }}/{{ .Values.api.image.repository }}:{{ .Values.api.image.tag }}"
+          imagePullPolicy: {{ .Values.api.image.pullPolicy | default .Values.image.pullPolicy }}
+          env:
+            - name: POSTGRES_USER
+              value: {{ .Values.api.env.postgres_user }}
+            - name: POSTGRES_PASSWORD
+              value: {{ .Values.api.env.postgres_password }}
+            - name: POSTGRES_HOST
+              value: {{ .Values.api.env.postgres_host }}
+            - name: POSTGRES_DB
+              value: {{ .Values.api.env.postgres_db }}
+            - name: POSTGRES_PORT
+              value: {{ .Values.api.env.postgres_port | quote }}
```

you can use default values
```suggestion
              value: {{ default "5432" .Values.api.env.postgres_port | quote }}
```

---

also I don't think the api.env part exists in the values file.

---

I think this is the api.env part, right? https://github.com/policy-design-lab/farmdoc-helm/blob/FD-518-create-helm-chart/values.yaml#L120

---

yup, just looked at the diffs.

---

added default parameter